library/constant_time_internal.h

```diff
@@ -221,6 +221,14 @@ void mbedtls_ct_memcpy_if_eq( unsigned char *dest,
  * offset_secret, but only on \p offset_min, \p offset_max and \p len.
  * Functionally equivalent to `memcpy(dst, src + offset_secret, len)`.
  *
+ * \note                In order to remain constant time this function reads
+ *                      from the \p dest buffer as well as writing to it, thus
+ *                      if used on an uninitialised buffer, this may generate
+ *                      static analysis warnings. Nothing is done with the
+ *                      read data however, other than putting it back if the
+ *                      condition is false, so this is not undefined
+ *                      behaviour.
```

Comment:
(nit pick) it's not UB because reading uninitialised data is only UB under very specific conditions.

the best I can come up with is

```
 *                      if used on an uninitialised buffer, this may generate
 *                      static analysis warnings (note: program execution is
 *                      not impacted by any such undefined values read, so this
 *                      is a false positive, and not UB in any case).
```

Reply:
To nitpick the nitpick, the C standard uses “unspecified values” — it uses the adjective “undefined” only when anything can happen (“undefined behavior”), whereas unspecified value/behavior means that there is a well-defined set of possible behaviors (in the case of values: a well-defined set of values, namely, the value has a well-defined number of bits and each bit of the value can be either 0 or 1) but the standard does not specify which behavior/value is chosen.

I suggest making the note shorter and more to the point.

> This function reads from `\p dest`, but the value that is read does not influence the result and this function's behavior is well-defined regardless of the contents of the buffers. This may result in false positives from static or dynamic analyzers, especially if `\p dest` is not initialized.